src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsFileProvider.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore
+{
+    /// <summary>
+    /// A <see cref="IFileProvider"/> for serving static web assets during development.
+    /// </summary>
+    internal class StaticWebAssetsFileProvider : IFileProvider
```

Comment:
All these classes will go into a separate package. Adding them to the default builder for now as we want to avoid delaying the feature from preview6 and all these are an implementation detail

Reply:
Why does this justify a separate package?

Reply:
Maybe not. I just didn’t want to cobble it together in here as this package is meant to be absolutely minimal.

Reply:
Are these ever going to be called from outside of CreateDefaultBuilder? If so then they need a different home.

Adding a new package is not a significant cost. Ask @aspnet/build if you need help.

Reply:
Alternatively these could be added to Hosting.

Reply:
I wouldn't move these to hosting, as they are kind of an "opinion" about how we do things. Lets leave them here for now as they are an implementation detail. We can move them to hosting in future without causing a breaking change as they are all internal.

(I believe we could even typeforward the class with the extension methods if we wanted to)

Reply:
The only opinion here is the line in ConfigureWebDefaults that adds this by default. Everything else should be moved elsewhere. This assembly does not provide features, only opinions.

Reply:
@Tratcher Fair enough. I'm going to be in Redmond next week, so lets leave it here for now and discuss it at that time. I don't want to be creating a package 2 days from code complete and this is an implementation detail that can be handled at the beginning of preview7.

Does that sound right?

Reply:
I'll file an issue for it.

Reply:
https://github.com/aspnet/AspNetCore/issues/10656